subsys/bluetooth/controller/ll_sw/nordic/lll/lll_test.c

```diff
@@ -193,74 +499,119 @@ static uint32_t init(uint8_t chan, uint8_t phy, void (*isr)(void *))
 	/* NOTE: No whitening in test mode. */
 	radio_phy_set(test_phy, test_phy_flags);
 	radio_tmr_tifs_set(150);
-	radio_tx_power_max_set();
+
+	err = tx_power_set(tx_power);
+
 	radio_freq_chan_set((chan << 1) + 2);
 	radio_aa_set((uint8_t *)&test_sync_word);
 	radio_crc_configure(0x65b, PDU_AC_CRC_IV);
-	radio_pkt_configure(RADIO_PKT_CONF_LENGTH_8BIT, 255, RADIO_PKT_CONF_PHY(test_phy));
-
-	return 0;
+	radio_pkt_configure(RADIO_PKT_CONF_LENGTH_8BIT, 255,
+			    RADIO_PKT_CONF_PHY(test_phy) |
+			    RADIO_PKT_CONF_PDU_TYPE(IS_ENABLED(CONFIG_BT_CTLR_DF_CTE_TX) ?
+								RADIO_PKT_CONF_PDU_TYPE_DC :
+								RADIO_PKT_CONF_PDU_TYPE_AC) |
+			    RADIO_PKT_CONF_CTE(cte ? RADIO_PKT_CONF_CTE_ENABLED :
+						     RADIO_PKT_CONF_CTE_DISABLED));
+
+	return err;
 }
 
-uint32_t ll_test_tx(uint8_t chan, uint8_t len, uint8_t type, uint8_t phy)
+static void payload_set(uint8_t type, uint8_t len, uint8_t cte_len, uint8_t cte_type)
 {
-	uint32_t start_us;
-	uint8_t *payload;
-	uint8_t *pdu;
-	uint32_t err;
+	struct pdu_dtm *pdu = radio_pkt_scratch_get();
 
-	if ((type > 0x07) || !phy || (phy > 0x04)) {
-		return BT_HCI_ERR_UNSUPP_FEATURE_PARAM_VAL;
-	}
-
-	err = init(chan, phy, isr_tx);
-	if (err) {
-		return err;
-	}
+	pdu->type = type;
+	pdu->length = len;
 
-	tx_req++;
 
-	pdu = radio_pkt_scratch_get();
-	payload = &pdu[2];
+#if defined(CONFIG_BT_CTLR_DF_CTE_TX)
+	pdu->cp = cte_len ? 1 : 0;
+	pdu->cte_info.time = cte_len;
+	pdu->cte_info.type = cte_type;
+#else
+	ARG_UNUSED(cte_len);
+	ARG_UNUSED(cte_type);
+#endif /* CONFIG_BT_CTLR_DF_CTE_RX */
 
 	switch (type) {
 	case 0x00:
-		memcpy(payload, prbs9, len);
+		memcpy(pdu->payload, prbs9, len);
 		break;
 
 	case 0x01:
-		memset(payload, 0x0f, len);
+		memset(pdu->payload, 0x0f, len);
 		break;
 
 	case 0x02:
-		memset(payload, 0x55, len);
+		memset(pdu->payload, 0x55, len);
 		break;
 
 	case 0x03:
-		memcpy(payload, prbs15, len);
+		memcpy(pdu->payload, prbs15, len);
 		break;
 
 	case 0x04:
-		memset(payload, 0xff, len);
+		memset(pdu->payload, 0xff, len);
 		break;
 
 	case 0x05:
-		memset(payload, 0x00, len);
+		memset(pdu->payload, 0x00, len);
 		break;
 
 	case 0x06:
-		memset(payload, 0xf0, len);
+		memset(pdu->payload, 0xf0, len);
 		break;
 
 	case 0x07:
-		memset(payload, 0xaa, len);
+		memset(pdu->payload, 0xaa, len);
 		break;
 	}
 
-	pdu[0] = type;
-	pdu[1] = len;
-
 	radio_pkt_tx_set(pdu);
+}
+
+uint32_t ll_test_tx(uint8_t chan, uint8_t len, uint8_t type, uint8_t phy,
+		    uint8_t cte_len, uint8_t cte_type, uint8_t switch_pattern_len,
+		    const uint8_t *ant_id, int8_t tx_power)
+{
+	uint32_t start_us;
+	uint32_t err;
+	bool cte_request = (cte_len > 0) ? 1 : 0;
+
+	if ((type > 0x07) || !phy || (phy > 0x04)) {
```

Comment:
Change magic numbers to macros.

Reply:
This isn't a part of this PR but I defined most of the magic number in this source file now